src/Core/Tool/Uploader.php

```diff
@@ -44,9 +44,10 @@ public function upload(UploadData $file, $filename = null)
 			$filename = uniqid() . '-' . $file->name;
 		}
 
-		// Avoid any possible issues with case sensitivity by forcing all files
-		// to be made lowercase.
-		$filename = strtolower($filename);
+		// Avoid possible issues with case sensitivity by forcing all files
+		// to be made lowercase. Also replace possibly invalid characters in filename
+		$filename = strtolower(preg_replace('/[^\pL\pN\-\_\s\.]+/u', '', $filename));
```

Comment:
this made my brain hurt, but it seems to be correct. Just wondering...are we planning to use the same for lumen?

Reply:
The regex is actually straight forward but the character classes are confusing: `\p` and `\N` are shorthand for all unicode characters and numbers, not just latin ones. So it replaces everything that isn't letters, number - _ or whitespace